### ISSUE TITLE
String Slice fails with 'Index was outside the bounds of the array.' error

### ISSUE THREAD

**Issue by hell-racer** [jdevillard/JmesPath.Net#88]
The following test would fail:

```
        [Theory]
        [InlineData("[:30]", "'abc'", "'abc'")]
        public void JmesPathSliceExpression_Transform_String(string slices, string input, string expected)
            => Assert(slices, input, expected);
```

This is because string slice doesn't have bounds check.

https://github.com/jdevillard/JmesPath.Net/blob/master/src/jmespath.net/Expressions/JmesPathSliceProjection.cs#L85-L86

```
            for (var index = start; compare(index, stop); index += step)
                characters.Add(text.CodePoints[index]);
```

**Comment by springcomp** [comment]
@hell-racer is this solved ? I seem to remember having addressed this in one of the latest pull requests ?

**Comment by hell-racer** [comment]
I think this one is fixed with #93.

**Comment by springcomp** [comment]
Your feedback is much appreciated.
Thank you.
